pkg/virt-controller/watch/migration/migration.go

```diff
@@ -1069,6 +1078,39 @@ func (c *Controller) getNodeSelectorsFromNodeName(nodeName string) (map[string]s
 	return res, nil
 }
 
+// updateTargetPodNetworkInfo generates the network-info annotation from the target pod's network-status
+func (c *Controller) updateTargetPodNetworkInfo(vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod) error {
```

Comment:
I think we should aim for a more general logic, following the pattern used in the VMI controller [1]. This will help us avoid coupling the core logic with network-specific logic and dependencies.

[1] https://github.com/kubevirt/kubevirt/blob/4a36c82d427f2a2b05ca2377de448693888a6d06/pkg/virt-controller/watch/vmi/lifecycle.go#L718

Reply:
> Is it safe to update k8s.v1.cni.cncf.io/networks during migration ? maybe it will uncover some corner case bugs ?

It can potentially happen today, at least to the source virt-launcher.
This annotation can take affect on two occasions:
1. Upon pod creation (in our case, the migration target pod).
2. In case the dynamic-networks-controller is deployed - it can trigger Multus to add / remove interfaces.

I think our flows are robust enough to handle it, but if you see a potential flaw - please raise it.